### DISCUSSION TITLE
EdgeX microservices are not running even after setting EDGEX_SECURITY_SECRET_STORE=false

### DISCUSSION THREAD

**Discussion by vediyappan-villali** [edgexfoundry/community#66] [Q&A]
Earlier, I was able to run the EdgeX microservices by making EDGEX_SECURITY_SECRET_STORE=false. But it is not working now. Should I do any other configuration changes.

```
> vediyappan@LX-9N9H9R2:~/work/libraries/tmp/edgex-go/cmd/core-data$ export EDGEX_SECURITY_SECRET_STORE=false && ./core-data
> level=INFO ts=2023-03-07T10:23:56.157786917Z app=core-data source=config.go:503 msg="Loaded private configuration from ./res/configuration.toml"
> level=INFO ts=2023-03-07T10:23:56.157816538Z app=core-data source=config.go:177 msg="Using local private configuration from file"
> level=INFO ts=2023-03-07T10:23:56.158351533Z app=core-data source=config.go:196 msg="Configuration loaded with 0 overrides applied"
> level=WARN ts=2023-03-07T10:23:56.158366445Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: Error, secretName () doesn't exist in secret store"
> level=WARN ts=2023-03-07T10:23:57.15864589Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: Error, secretName () doesn't exist in secret store"
> level=WARN ts=2023-03-07T10:23:58.159137711Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: Error, secretName () doesn't exist in secret store"
> level=WARN ts=2023-03-07T10:23:59.159622675Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: Error, secretName () doesn't exist in secret store"
> ^Clevel=WARN ts=2023-03-07T10:24:00.16024866Z app=core-data source=database.go:91 msg="couldn't retrieve database credentials: Error, secretName () doesn't exist in secret store"
> ^C
> vediyappan@LX-9N9H9R2:~/work/libraries/tmp/edgex-go/cmd/core-data$

```

**Comment by iain-anderson** [comment]
Not sure about a non-consul setup, but at the moment I'm doing
`docker run -p 8500:8500 consul`
in one window, and then
```
declare -x EDGEX_SECURITY_SECRET_STORE=false
cd cmd/core-common-config-bootstrapper
./core-common-config-bootstrapper -cp -cf configuration.yaml
cd ../core-metadata
./core-metadata -cp
```
in another (the config bootstrapper only needs to be run once unless you clean out consul)

**Reply by vediyappan-villali** [reply]
@iain-anderson Actually I am running the natively build EdgeX microservices in Linux environment (Ubuntu 22.04). Not using the docker and consul setup.

**Reply by iain-anderson** [reply]
Same here, consul is the only thing I'm running in a container

**Comment by khalilesisar** [comment]
Hello , did you find a solution please ? i have the same problem

**Comment by bnevis-i** [comment]
Here's where this code is failing:

https://github.com/edgexfoundry/edgex-go/blob/main/internal/pkg/bootstrap/handlers/database.go#L91

and here is the line in configuration.toml that contains the secret name that the code is trying to get:

https://github.com/edgexfoundry/edgex-go/blob/jakarta/cmd/core-data/res/configuration.toml#L46


Why is Databases.Primary.Type = ""?  You might have to dive into the debugger to figure it out.

**Comment by dipshirajput** [comment]
https://github.com/orgs/edgexfoundry/discussions/66#discussioncomment-5227446
Hi @vediyappan-villali 
are you still facing the same error? are you running redis before you run core-data ? as you will need to use redis. and also can you post your configuration.toml for core-data you are using so that I can help you out in a better way ?

**Reply by vediyappan-villali** [reply]
@dipshirajput Yes, I am running redis before running core-data as I need to use redis. I am using the configuration.toml file from repo without any changes. Link is as follows:
https://github.com/edgexfoundry/edgex-go/blob/main/cmd/core-data/res/configuration.toml

**Comment by lenny-goodell** [comment]
@vediyappan-villali , what version of EdgeX are you running. 

Please note that the `main` branch is the **development** branch for EdgeX and it is not recommend to use unless you are part of the current development effort. This is very important during this release development cycle as there are many breaking changes occurring.
